packages/dashmate/src/listr/tasks/reindexNodeTaskFactory.js

```diff
@@ -38,125 +64,99 @@ function reindexNodeTaskFactory(
 
-          // Write configs
-          configFileRepository.write(configFile);
-          const configFiles = renderServiceTemplates(config);
-          writeServiceConfigs(config.getName(), configFiles);
-        },
-      },
-      {
-        title: 'Start core',
-        task: async (ctx) => {
-          const { docker } = dockerCompose;
-
-          let containerId = config.get('core.reindex.containerId', false);
-          let containerInfo;
-
-          if (containerId) {
-            try {
-              containerInfo = await docker.getContainer(containerId).inspect();
-            } catch (e) {
-              if (e.reason === 'no such container') {
-                containerId = null;
-              }
-              throw e;
-            }
-          }
+          const serviceConfig = renderTemplate(path.join(TEMPLATES_DIR, templatePath),
+            { ...config.options, reindex: true });
+          writeServiceConfigs(config.getName(), { [configPath]: serviceConfig });
 
-          if (!containerId) {
-            ctx.coreService = await startCore(config);
-            containerInfo = await ctx.coreService.dockerContainer.inspect();
+          const coreContainer = await getCoreContainer(config);
+          if (coreContainer) {
+            const info = await coreContainer.inspect();
 
-            ctx.reindexContainerId = containerInfo.Id;
-            config.set('core.reindex.containerId', containerInfo.Id);
-            configFileRepository.write(configFile);
-
-            return;
+            if (info.State.Status !== 'exited') {
+              await coreContainer.restart();
+              return Promise.resolve();
+            }
+            await coreContainer.start();
+            return Promise.resolve();
           }
+          const isNodeRunning = await dockerCompose.isNodeRunning(generateEnvs(configFile, config));
 
-          ctx.reindexContainerId = containerInfo.Id;
-          ctx.coreService = new CoreService(
-            config,
-            createRpcClient(
-              {
-                port: config.get('core.rpc.port'),
-                user: config.get('core.rpc.user'),
-                pass: config.get('core.rpc.password'),
-                host: await getConnectionHost(config, 'core'),
-              },
-            ),
-            docker.getContainer(containerId),
-          );
-
-          const { State } = containerInfo;
-
-          if (State.Status === 'paused' || State.Status === 'exited') {
-            switch (State.ExitCode) {
-              default:
-              // eslint-disable-next-line no-console
-                console.warn(`Reindex container exited with status ${State.ExitCode}, check docker logs of container ${containerId}`);
-              // eslint-disable-next-line no-fallthrough
-              case 0:
-                await docker.getContainer(containerId).start();
-            }
+          if (isNodeRunning) {
+            return restartNodeTask(config);
```

Comment:
Why do we restart the whole node but not just the core service? Please add more comments to this function to explain logic and reasoning.